### PR TITLE
device selector listens to device notifications

### PR DESCRIPTION
**Pull request by devoncarew**
- have the device selector listen to device notifications
- auto-select a device if we find one and there is no current selection
- on the mac, show a menu item for opening the ios simulator

screen shot 2016-09-29 at 11 21 14 pm

screen shot 2016-09-29 at 11 21 27 pm

@pq @stevemessick

### CODE REVIEW

**Comment by devoncarew** [comment]
fix https://github.com/flutter/flutter-intellij/issues/179, https://github.com/flutter/flutter-intellij/issues/182

File: src/io/flutter/run/daemon/FlutterDaemonService.java [hunk 1]
```diff
@@ -94,7 +104,9 @@ private FlutterDaemonService() {
   /**
    * @return the currently selected device
    */
-  public @Nullable ConnectedDevice getSelectedDevice() {
+  public
+  @Nullable
+  ConnectedDevice getSelectedDevice() {
```
Comment (stevemessick) [inline src/io/flutter/run/daemon/FlutterDaemonService.java]: We usually put access on the declaration line.

File: src/io/flutter/actions/DeviceSelectorAction.java [hunk 2]
```diff
@@ -20,27 +32,36 @@
...
+
+    if (actions.isEmpty()) {
+      group.add(new NoDevicesAction());
+    }
+    else {
+      group.addAll(actions);
+    }
+
+    if (SystemInfo.isMac) {
+      boolean simulatorOpen = false;
```
Comment (stevemessick) [inline src/io/flutter/actions/DeviceSelectorAction.java]: Not used?

**Comment by eseidelGoogle** [comment]
"..." used to imply "opens a dialog" in Apple's HUG, no clue if that's
still the case, but my instinct is you don't want the ... here.

File: src/io/flutter/run/daemon/FlutterDaemonService.java (hunk 1 above)
Comment (devoncarew) [inline src/io/flutter/run/daemon/FlutterDaemonService.java]: done

File: src/io/flutter/actions/DeviceSelectorAction.java (hunk 2 above)
Comment (devoncarew) [inline src/io/flutter/actions/DeviceSelectorAction.java]: We pass it into the `OpenSimulatorAction` ctor, int order to toggle enablement of the action.

**Comment by devoncarew** [comment]
Re: the ellipses my initial impl had show a toast on success. I'll experiment with showing progress as the simulator opens, and remove the ellipses if that doesn't work out.

**Comment by pq** [comment]
\o/

**Pull request merged**